language/move-vm/runtime/src/interpreter.rs

```diff
@@ -400,7 +400,7 @@ impl<L: LogContext> Interpreter<L> {
         if err.status_type() == StatusType::Verification {
             self.log_context.alert();
             error!(
-                self.log_context,
+                *self.log_context.as_super(),
```

Comment:
I'm surprised the error macro can't handle this directly. i.e. it should be able to handle anything that implements `Schema`

Reply:
The fundamental issue here is Rust's inability to coerce `&dyn Foo` to `&dyn Bar` where `trait Foo: Bar`, unlike other (OOP) languages due to its choice of vtable layouts. 

https://users.rust-lang.org/t/casting-traitobject-to-super-trait/33524

Reply:
I think the bigger problem here however is that `trait LogContext` (a Move VM concept) should not have `Schema` (from Diem logging) as a super trait since it's a violation of the Move/Diem split.